src/main/java/org/teamresistance/frc/OpticalFlow.java

```diff
@@ -0,0 +1,149 @@
+package org.teamresistance.frc;
+
+import edu.wpi.first.wpilibj.SPI;
+import edu.wpi.first.wpilibj.SPI.Port;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+/**
+ * Created by Joseph for testing the Optical Flow
+ *
+ * Registers
+ * 0  -   Product ID
+ * 1  -   Revision ID
+ * 2  -   Motion
+ * 3  -   Delta x
+ * 4  -   Delta y
+ * 5  -   Squall
+ *
+ */
+public class OpticalFlow {
+
+  private SPI spi;
+  private  byte[] dataReceived;
+  private  byte[] register = new byte[] {0};
+
+  //Amount of change since last read
+  double raw_dx;
+  double raw_dy;
+
+  //Summation of raw_dx/y over time
+  double tot_dx;
+  double tot_dy;
+
+  //Pixel to Feet conversion factor
+  double xFacLeftFt = 750;
+  double xFacRightFt = 880;
+  double yFacFwdFt = 780;
+  double yFacRevFt = 880;
+
+  //Distance Covered
+  double dx;
+  double dy;
+
+  //Vars for Franks code
+//  double xFacFt = 800;
+//  double yFacFt = 800;
+
+
+  public OpticalFlow() {
+    spi = new SPI(Port.kOnboardCS0);    //Finds the OF on the SPI ports
+    spi.setChipSelectActiveLow();
+    spi.setClockActiveHigh();
+    spi.setClockRate(500000);
+
+    dataReceived = new byte[1];
+    for(int i = 0; i < dataReceived.length; i++) {
+      dataReceived[i] = 0;
+    }
+  }
+
+  public void init() {
+    int motionRegister = readRegister((byte)2);
+    SmartDashboard.putNumber("Motion Register", motionRegister);
+    if((motionRegister & 0x80) != 0) {
```

Comment:
If I understand what @jaxfrank  told me correctly, then this is a procedure that needs to run in order to make the OF run, neither one of us knows why.